extensions/amp-analytics/0.1/events.js

```diff
@@ -400,6 +410,152 @@ export class ClickEventTracker extends EventTracker {
   }
 }
 
+/**
+ * Tracks scroll events.
+ */
+export class ScrollEventTracker extends EventTracker {
+  
+  /**
+   * @param {!./analytics-root.AnalyticsRoot} root
+   */
+  constructor(root) {
+    super(root);
+
+    /** @private {!Observable<!../../../src/service/viewport/viewport-impl.ViewportChangedEventDef>} */
+    this.scrollObservable_ = new Observable();
+
+    /** @const @private {!../../../src/service/viewport/viewport-impl.Viewport} */
+    this.viewport_ = Services.viewportForDoc(root.ampdoc);
+    
+    /** @private {!./analytics-root.AnalyticsRoot} root */
+    this.root_ = root;
+  }
+
+  /** @override */
+  dispose() {
+    this.scrollObservable_.removeAll();
+  }
+
+  /** @override */
+  add(context, eventType, config, listener) {
+    if (!config['scrollSpec']) {
+      user().error(TAG, 'Missing scrollSpec on scroll trigger.');
+      return;
+    }
+    this.registerScrollTrigger_(config['scrollSpec'], listener);
+
+    // Trigger an event to fire events that might have already happened.
+    const size = this.viewport_.getSize();
+    this.onScroll_({
+      top: this.viewport_.getScrollTop(),
+      left: this.viewport_.getScrollLeft(),
+      width: size.width,
+      height: size.height,
+      relayoutAll: false,
+      velocity: 0, // Hack for typing.
+    });
+  }
+
+  /**
+   * Register for a listener to be called when the boundaries specified in
+   * config are reached.
+   * @param {!JsonObject} config the config that specifies the boundaries.
+   * @param {function(!AnalyticsEvent)} listener
+   * @private
+   */
+  registerScrollTrigger_(config, listener) {
+    if (!Array.isArray(config['verticalBoundaries']) &&
+      !Array.isArray(config['horizontalBoundaries'])) {
+      user().error(TAG, 'Boundaries are required for the scroll ' +
+        'trigger to work.');
+      return;
+    }
+
+    // Ensure that the scroll events are being listened to.
+    if (!this.scrollHandlerRegistered_) {
```

Comment:
`this.scrollHandlerRegistered_` is not set to `true` anywhere. After the migration I also don't expect the `#registerScrollTrigger_` to be called more than once within the same instance.
We used to have `scrollHandlerRegistered_` because we don't want to add two listener to the viewport scroll event. and that's mostly due to we don't want to normalized twice if there are two `scroll` event listener. 
To achieve this, you will need to move the variable `this.scrollHandlerRegistered_` outside this instance scope and maybe have the analytics root keep a record for you. 

I am expecting a similar `scrollHandler` that can be shared across multiple `ScrollEventTracker ` instance. 
Please refer to the [visibilityManager](https://github.com/ampproject/amphtml/blob/master/extensions/amp-analytics/0.1/analytics-root.js#L329) that we have.

Reply:
Hello,

Thanks for this! I honestly was going to remove `this.scrollHandlerRegistered_`. but I guess I left that that dangling on accident. and now I am glad I did haha!

And awesome, I'll go ahead and implement that. 😄